etl/loader.py

```diff
@@ -35,37 +38,40 @@ def connect(self):
         # Get credentials from YAML
         credentials = self.get_credentials(self._config,'database_credentials')
         # create sqlalchemy engine
-        engine = sqlalchemy.create_engine("mysql+pymysql://{user}:{pw}@{host}/{db}"
-                               .format(user = credentials['db_username'],
-                                       host = credentials['db_hostname'],
-                                       pw = credentials['db_password'],
-                                       db = credentials['db_name']))
+        engine = sqlalchemy.create_engine("{dialect}://{user}{pw}{host}/{db}"
+                               .format(dialect = xstr(credentials['dialect_driver']),
+                                        user = xstr(credentials['db_username']),
+                                        host = ('@', '')[credentials['db_hostname'] is None] + xstr(credentials['db_hostname']),
+                                        pw = (':', '')[credentials['db_password'] is None] + xstr(credentials['db_password']),
+                                        db = xstr(credentials['db_name'])))
         # try to connect to database
         try:
             engine.connect()
             self._engine = engine
+            self._metadata = sqlalchemy.MetaData()
         except sqlalchemy.exc.OperationalError as except_detail:
             # print error details
             print("ERROR: {}".format(except_detail))
         return
 
-    def insert(self, dataframe, table_name, chunksize):
+    def insert(self, df_dict):
```

Comment:
We will need match up the transformer output and loader input. The transformer outputs dict['transform task name'] = dataframe but the loader expects dict['table name']. At some point we should either have the transformer output using table name as key, or have the transformer task config contain a table name.

Reply:
Changed insert() to insert only one table (dataframe). Put the transform_dict looping in `appp.py`